src/backend/distributed/sql/udfs/citus_job_status/latest.sql

```diff
@@ -0,0 +1,98 @@
+CREATE OR REPLACE FUNCTION pg_catalog.citus_job_status (
+    job_id bigint,
+    raw boolean DEFAULT FALSE
+)
+    RETURNS TABLE (
+            job_id bigint,
+            state pg_catalog.citus_job_status,
+            job_type name,
+            description text,
+            started_at timestamptz,
+            finished_at timestamptz,
+            details jsonb
+    )
+    LANGUAGE SQL
+    STRICT
+    AS $fn$
+    WITH rp AS MATERIALIZED (
+        SELECT
+            sessionid,
+            sum(source_shard_size) as source_shard_size,
+            sum(target_shard_size) as target_shard_size,
+            any_value(status) as status,
+            any_value(sourcename) as sourcename,
+            any_value(sourceport) as sourceport,
+            any_value(targetname) as targetname,
+            any_value(targetport) as targetport,
+            max(source_lsn) as source_lsn,
+            min(target_lsn) as target_lsn
+        FROM get_rebalance_progress()
+        GROUP BY sessionid
+    ),
+    task_state_occurence_counts AS (
+        SELECT t.status, count(task_id)
+        FROM pg_dist_background_job j
+            JOIN pg_dist_background_task t ON t.job_id = j.job_id
+        WHERE j.job_id = $1
+        GROUP BY t.status
+    ),
+    running_task_details AS (
+        SELECT jsonb_agg(jsonb_build_object(
+                'state', t.status,
+                'retried', coalesce(t.retry_count,0),
+                'phase', rp.status,
+                'size' , jsonb_build_object(
+                    'source', rp.source_shard_size,
+                    'target', rp.target_shard_size),
+                'hosts', jsonb_build_object(
+                    'source', rp.sourcename || ':' || rp.sourceport,
+                    'target', rp.targetname || ':' || rp.targetport),
+                'message', t.message,
+                'command', t.command,
+                'LSN', jsonb_build_object(
+                    'source', rp.source_lsn,
+                    'target', rp.target_lsn),
```

Comment:
Based on the docs I think adding the lag is probably as simple as:
```suggestion
                    'target', rp.target_lsn
                    'lag', rp.source_lsn - p.target_lsn),
```
(and using pg_size_pretty based on value of `raw`)

Reply:
It was slightly more complicated than this as we need to coalesce null pg_lsn values now, but it is ok.

I did not really know how to read the lsn lag and it was a bit confusing for me but I assume it would make more sense to an experienced operator.

```json
{
  "LSN": {
    "lag": "1773 MB",
    "source": "0/6ED36D00",
    "target": "0/0"
  },
  "size": {
    "source": "140 MB",
    "target": "132 MB"
  }
}
```

In my test setup, I added a second worker node and started a rebalance in the background. The sum of total disk size of all distributed tables is less than 1500 MB. The amount of data to move to the new shard is roughly half of this: 750 MB. Seeing a replication lag of 1773 MB here looks like there is more data to move around than I anticipated. Clearly this may lead to confusion in people such as me, who did not understand the concept of replication lag fully.

Reply:
Thanks for the explanation. Not it all makes sense. I rebased and updated the commit [0738d46](https://github.com/citusdata/citus/pull/6576/commits/0738d464e170bf0feb46ff9e44058ece1186ac97) that contains changes for lsn lag